### DISCUSSION TITLE
What's in-scope?

### DISCUSSION THREAD

**Discussion by jcbhmr** [jimmywarting/web-worker-polyfill#18] [General]
There's other stuff that is specified in the https://html.spec.whatwg.org/multipage/workers.html page relating to how the `Worker` interface works. For instance, there's a `WorkerNavigator` and `WorkerLocation`. I assume these are out-of-scope, but I just want to confirm? There's also the `onoffline`, `onlanguagechange`, and `ononline` events.

![image](https://github.com/jimmywarting/whatwg-worker/assets/61068799/fcd83193-ada8-4aa8-b509-725b42e6eb4c)

**Comment by jcbhmr** [comment]
Another thing I've noticed is that there appears to be support for importing `http:` and `https:` URLs. Is this something that can be relegated to `--experimental-network-imports`? It appears that it's already native, just behind a flag!
https://nodejs.org/api/esm.html#https-and-http-imports

Also consider that some people may not want to have HTTP imports.
If you _really_ want to use a custom HTTP loader, I suggest having a conditional like:

```js
if (process.execArgv.includes("--experimental-network-imports") {
  applyCustomHTTPLoader(); // Or whatever...
}
```

Thoughts? 🤔

**Comment by jimmywarting** [comment]
The hole reason i wanted to have custom experimental loaders where not particularly import `http` url. 
it was so that i could create dynamic code and import them with blob + ObjectURL (for await-sync)

so the idea was to do something like:

```js
const blob = new Blob([`
  import fs from 'node:fs'
  fs.writeFileSync(...)
`], { type: 'text/javascript' })

const url = URL.createObjectURL(blob)
import(url)
```

But sure this could maybe be an option for this. 
but i maybe want to avoid this for now just to get out a working prototype and creating a MVP

**Comment by jimmywarting** [comment]
There are some thing that NodeJS is working on about implementing `globalThis.navigator.userAgent` with push from wintercg minium api: https://common-min-api.proposal.wintercg.org/

there are issues here as well: 
- https://github.com/nodejs/node/issues/39540
- https://github.com/nodejs/node/pull/47769 


but we could implement whatever we feel like a web worker should actually have

**Comment by jcbhmr** [comment]
I just realized Discussions were enabled on this repo. 🤦‍♂️ If you could, would move this issue to a Discussion pls to satisfy my brain 🧠🙏

**Comment by jimmywarting** [comment]
I just tough that i do not particularly want to ship just about everything that should exist inside a Web Worker. 
i don't want this to end up bing one of those bloated libraries that ends up being a hole https://polyfill.io thing. 

cuz if it where then i would just probably just change [await-sync](https://github.com/jimmywarting/await-sync) to just inline some functionality and work with different code paths to deal with things if they are either NodeJS or running inside of a browser or deno. 
I don't want it to come with the cost of having to depend on eg the hole undici dependency, so so the worker could have fetch builtin to v16. 

i would kind of like it if there where a way to add optional dependencies to the library. and that we tried to load `node-fetch` or `undici` only if it's added by the users themself and if they really need it.
